tests/test_zoneinfo.py

```diff
@@ -1233,14 +1305,14 @@ def test_dir_unique(self):
         self.assertCountEqual(module_dir, module_unique)
 
 
-@dataclasses.dataclass
+@dataclasses.dataclass(frozen=True)
```

Comment:
I assume this is a change that could happen even without the C extension... it *might* be worth separating improvements like this out (possibly as a group) so that they can go in sooner, and this pull will be slightly smaller.